Add restore process (#3562)
sourcegraph/handbook: content/departments/product-engineering/engineering/cloud/devops/managed/restore_process.md

# Restore a managed instance when it is completely gone

<span class="badge badge-note">SOC2/CI-110</span>

> ## Assess what is deleted

Reviewer: @michaellzc for v1.1 we also need to add CloudSQL restore process - should I extend your PR? Or do it in separate?

Also for [SOC2-111](https://github.com/sourcegraph/security-issues/issues/247) I need to perform all actions from this page, write date-time and prove we do it annually - I will do it in google doc and ask you to review, ok?

Author: Hey @filiphaftek 

there's a separate thread where I was trying to get more clarify on what's required https://sourcegraph.slack.com/archives/C02Q9G9A59S/p1653334149100629

but yeah.

I am updating this PR to include process to restore Cloud SQL for MI v1.1 as well.

> Also for https://github.com/sourcegraph/security-issues/issues/247 I need to perform all actions from this page, write date-time and prove we do it annually - I will do it in google doc and ask you to review, ok?

yes please, thank you!

# Restore a managed instance when it is completely gone

<span class="badge badge-note">SOC2/CI-110</span>

**For MI 1.1, see the new process [here](v1.1/mi1-1_restore_process.md).**

## Assess what is deleted

Navigate to `sourcegraph-managed-$CUSTOMER` project, and look at existing compute instance. Does the VM still exist?

- No, the VM is gone.

  - Does the data disk `default-$CURRENT_DEPLOYMENT-data-disk` still exists?
    - Yes, [re-create the VM with existing data disk](##re-create-the-vm-with-existing-data-disk)
    - No, [re-create the VM with a new data disk from disk snapshot](##re-create-the-vm-with-new-data-disk-from-disk-snapshot)

> - Yes, follow [operation guides](./operations.md) to troubleshoot services condition. If unable to recover running services on the VM, fallback to the recovery plan above.

Reviewer 2: > fallback to the recovery plan above.

Maybe specify which step specifically - I presume it would be 1. recreate the VM, then 2. recreate the VM with snapshot.

## Re-create the VM with existing data disk

1. Open [sourcegraph/deploy-sourcegraph-managed] and check out to the `$CUSTOMER` directory
1. Run `terraform apply` to reconcile the infrastructure to its definition in code.
1. Follow [confirm instance health](./operations.md#confirm-instance-health)

## Re-create the VM with new data disk from disk snapshot

> 1. Go to https://console.cloud.google.com/compute/snapshots (make sure you select the right project) and find the latest snapshot, note the name

Reviewer 2: ```suggestion
1. Run `gcloud compute snapshots list --project=sourcegraph-managed-$CUSTOMER --sort-by="~creationTimestamp" --limit=5 --format="table(name,creationTimestamp)"` and copy the name of the latest snapshot
```
